components/style/values/specified/mod.rs

```diff
@@ -621,19 +621,15 @@ impl Shadow {
             if !lengths_parsed {
                 if let Ok(value) = input.try(|i| Length::parse(context, i)) {
                     lengths[0] = value;
-                    let mut length_parsed_count = 1;
-                    while length_parsed_count < length_count {
+
+                    lengths[1] = try!(Length::parse(context, input));
+
+                    if let Ok(value) = input.try(|i| Length::parse_non_negative(i)) {
+                        lengths[2] = value;
+
```

Comment:
I wonder do we really need so many empty lines... Could you remove all the three empty lines you add above?